tests/include/mir/test/gmock_fixes.h

```diff
@@ -77,9 +77,9 @@ class ActionResultHolder<std::unique_ptr<T, D>>
   // result in a new-ed ActionResultHolder.
   template <typename F>
   static ActionResultHolder* PerformDefaultAction(
-      const FunctionMockerBase<F>* func_mocker,
+      const FunctionMocker<F>* func_mocker,
```

Comment:
This breaks the build on old GoogleTest.

Thankfully, it's not actually necessary. Please drop it :grin:

Reply:
I got this error when compiling without this PR: https://pastebin.com/raw/BJUaEcV3 . I then made this change and it got further until I hit something where I noticed that the GTEST_AT_LEAST macro was broken and discovered that the regex was broken. After I fixed that, the build went thorugh